plugins/dnc.js

```diff
@@ -0,0 +1,1404 @@
+        var feature = {};
+
+        for (var i=0, sLen = dnc.rawSchema.length; i < sLen; i++)
+        {
+            if (dnc.rawSchema[i].name == layerName)
+            {
+                feature = dnc.rawSchema[i];
+                break;
+            }
+        }
+
+        // Now validate the Enumerated values
+        for (var i=0, cLen = feature['columns'].length; i < cLen; i++)
+        {
+            // Skip non enumeratied attributes
+            if (feature.columns[i].type !== 'enumeration') continue;
+
+            var enumName = feature.columns[i].name;
+
+            // Skip stuff that is missing and will end up as a default value
+            if (!attrs[enumName]) continue;
+
+            var attrValue = attrs[enumName];
+            var enumList = feature.columns[i].enumerations;
+            var enumValueList = [];
+
+            // Pull all of the values out of the enumerated list to make life easier
+            for (var j=0, elen = enumList.length; j < elen; j++) enumValueList.push(enumList[j].value);
+
+            // Check if it is a valid enumerated value
+            if (enumValueList.indexOf(attrValue) == -1)
+            {
+                if (dnc.configOut.OgrDebugDumpvalidate == 'true') hoot.logWarn('Validate: Enumerated Value: ' + attrValue + ' not found in ' + enumName);
+
+                // Do we have an "Other" value?
+                if (enumValueList.indexOf('999') == -1)
+                {
+                    // No: Set the offending enumerated value to the default value
+                    attrs[enumName] = feature.columns[i].defValue;
+
+                    hoot.logTrace('Validate: Enumerated Value: ' + attrValue + ' not found in ' + enumName + ' Setting ' + enumName + ' to its default value (' + feature.columns[i].defValue + ')');
+                }
+                else
+                {
+                    // Yes: Set the offending enumerated value to the "other" value
+                    attrs[enumName] = '999';
+
+                    hoot.logTrace('Validate: Enumerated Value: ' + attrValue + ' not found in ' + enumName + ' Setting ' + enumName + ' to Other (999)');
+                }
+
+                // Since we either wiped the value or set it to '999', restore the tags for it 
+                if (enumName in transMap)
+                {
+                    notUsed[transMap[enumName][1]] = transMap[enumName][2];
+                    // Debug:
+                    print('Validate: re-adding enumeration ' + transMap[enumName][1] + ' = ' + transMap[enumName][2] + ' to notUsed');
+                }
+                else
+                {
+                    hoot.logError('Validate: ' + enumName + ' missing from transMap');
+                }
+            }
+        } // End Validate Enumerations
+    }, // End validateAttrs
+
+
+    // Apply one to one translations and don't report errors: missing columns etc
+    // THis does keep track of what has been used so we can undo it later if needed
```

Comment:
Just so that you know I'm looking at all of the code, `THis` doesn't need a capital `H`!  I don't care if you fix this or not...